exporter/sumologicexporter/prometheus_formatter.go

```diff
@@ -0,0 +1,445 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sumologicexporter
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+	"time"
+
+	"go.opentelemetry.io/collector/consumer/pdata"
+	tracetranslator "go.opentelemetry.io/collector/translator/trace"
+)
+
+type dataPointCommon interface {
+	Timestamp() pdata.TimestampUnixNano
+	LabelsMap() pdata.StringMap
+}
+
+type prometheusFormatter struct {
+	sanitNameRegex *regexp.Regexp
+}
+
+type prometheusTags string
+
+func newPrometheusFormatter() (prometheusFormatter, error) {
+	sanitNameRegex, err := regexp.Compile(`[^0-9a-zA-Z]`)
+	if err != nil {
+		return prometheusFormatter{}, nil
+	}
+
+	return prometheusFormatter{
+		sanitNameRegex: sanitNameRegex,
+	}, nil
+}
+
+// PrometheusLabels returns all attributes as sanitized prometheus labels string
+func (f *prometheusFormatter) tags2String(attr pdata.AttributeMap, labels pdata.StringMap) prometheusTags {
+	mergedAttributes := pdata.NewAttributeMap()
+	attr.CopyTo(mergedAttributes)
+	labels.ForEach(func(k string, v string) {
+		mergedAttributes.UpsertString(k, v)
+	})
+	length := mergedAttributes.Len()
+
+	if length == 0 {
+		return ""
+	}
+
+	returnValue := make([]string, 0, length)
+	mergedAttributes.ForEach(func(k string, v pdata.AttributeValue) {
+		returnValue = append(
+			returnValue,
+			fmt.Sprintf(
+				`%s="%s"`,
+				f.sanitizeKey(k),
+				f.sanitizeValue(tracetranslator.AttributeValueToString(v, false)),
+			),
+		)
+	})
+
+	return prometheusTags(fmt.Sprintf("{%s}", strings.Join(returnValue, ",")))
+}
+
+// sanitizeKey returns sanitized key string
+// (all non-alphanumeric chars replaced with `_`)
+func (f *prometheusFormatter) sanitizeKey(s string) string {
+	return f.sanitNameRegex.ReplaceAllString(s, "_")
+}
+
+// sanitizeKey returns sanitized value string
+// `/` -> `//`
+// `"` -> `\"`
+// `\n` -> `\n`
+func (f *prometheusFormatter) sanitizeValue(s string) string {
+	s = strings.ReplaceAll(s, `\`, `\\`)
+	s = strings.ReplaceAll(s, `"`, `\"`)
+	return strings.ReplaceAll(s, `\\n`, `\n`)
+}
+
+// doubleLine builds metric based on the given arguments where value is float64
+// this function shouldn't be called directly
```

Comment:
```suggestion
// doubleLine builds a metric based on the given arguments where value is float64.
```

> // This function shouldn't be called directly
Why? It's unexported so it's obvious that no one else should use it.

Reply:
attributes should be output of `tags2String`